### PR TITLE
adding a blog post for GitHub codespaces for teaching

### PR DESCRIPTION
**Pull request by Sparrow0hawk**
This PR adds a blog post and attributed thumbnail about using GitHub codespaces for teaching.

### CODE REVIEW

File: blog/_posts/2023-04-05-codespaces-teaching.md [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+
+A critical steep learning curve for both these courses is often set up. We encourage people to bring their own devices or use a University cluster machine and typically try and use tools like conda to help manage installation. This however adds additional complexity which feels like an extra cognitive burden for learners. Now they have to think about environments, directories to install things, etc etc. In recent courses we've accepted this and spent some time talking about these things in the time we’ve had available but for a 1-day course touching on both we wanted to try something a little different. For this course setup need to be easy so we didn’t have to think about installing Python or git, and be able to get straight into writing Python with a bit of git thrown in as well as integrating with GitHub.
```
Comment (patricia-ternes) [inline blog/_posts/2023-04-05-codespaces-teaching.md:20]: I have no idea why, but the apostrophe is being rendered differently through the text.
See the example with "we've" bellow

```
In recent courses we've accepted this and spent some time talking about these things in the time we’ve had...
```

File: blog/_posts/2023-04-05-codespaces-teaching.md [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+
+Our preference for teaching Python is to use a Jupyter notebook as the primary interface. This has its drawbacks but we accept these as instructors and try and reflect these during the session. GitHub codespaces  again makes this quite straightforward as VScode, the text editor codespaces is based on natively supports the Jupyter notebook format .ipynb. We can just create .ipynb files and specify that the kernel be the base Python installation and you can start to write notebooks. GitHub also supports (as a public beta) starting a codespace as a [Jupyter Lab session direct from GitHub](https://github.blog/changelog/2022-11-09-using-codespaces-with-jupyterlab-public-beta/). This requires that the codespace has JupyterLab installed but again is a nice option to immediately drop people into a notebook like environment similar to [Google Colab](https://colab.research.google.com/).
```
Comment (patricia-ternes) [inline blog/_posts/2023-04-05-codespaces-teaching.md:31]:
```suggestion
Our preference for teaching Python is to use a Jupyter notebook as the primary interface. This has its drawbacks but we accept these as instructors and try to reflect these during the session. GitHub codespaces  again makes this quite straightforward as VScode, the text editor codespaces is based on natively supports the Jupyter notebook format .ipynb. So, we can just create .ipynb files and specify that the kernel be the base Python installation and you can start to write notebooks. GitHub also supports (as a public beta) starting a codespace as a [Jupyter Lab session direct from GitHub](https://github.blog/changelog/2022-11-09-using-codespaces-with-jupyterlab-public-beta/). This requires that the codespace has JupyterLab installed but again is a nice option to immediately drop people into a notebook like environment similar to [Google Colab](https://colab.research.google.com/).
```

File: blog/_posts/2023-04-05-codespaces-teaching.md [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+
+Overall, GitHub Codespaces worked really nicely for what we (the instructors) wanted here. We hope that attendees also found it straightforward (pending feedback) but would definitely consider using this in future for our training sessions and encourage others who are looking for easy to configure, small compute environments for teaching to explore this offering.
+
+If you're interested in getting started with this I would recommend looking through the following resources:
+- GitHub documentation on [setting up a Python Codespace](https://docs.github.com/en/codespaces/setting-up-your-project-for-codespaces/adding-a-dev-container-configuration/setting-up-your-python-project-for-codespaces)
+- [Introduction to dev containers](https://docs.github.com/en/codespaces/setting-up-your-project-for-codespaces/adding-a-dev-container-configuration/introduction-to-dev-containers), the underlying technology for Codespaces
+- [GitHub Guide to creating a codespace on a repository](https://docs.github.com/en/codespaces/developing-in-codespaces/creating-a-codespace-for-a-repository)
```
Comment (patricia-ternes) [inline blog/_posts/2023-04-05-codespaces-teaching.md:38]: Linting: Just add a line before and after list (same as before)

```suggestion
- [GitHub Guide to creating a codespace on a repository](https://docs.github.com/en/codespaces/developing-in-codespaces/creating-a-codespace-for-a-repository)

```

File: blog/_posts/2023-04-05-codespaces-teaching.md (hunk 1 above)
Comment (Sparrow0hawk) [inline blog/_posts/2023-04-05-codespaces-teaching.md:20]:
```suggestion
A critical steep learning curve for both these courses is often set up. We encourage people to bring their own devices or use a University cluster machine and typically try and use tools like conda to help manage installation. This however adds additional complexity which feels like an extra cognitive burden for learners. Now they have to think about environments, directories to install things, etc etc. In recent courses we've accepted this and spent some time talking about these things in the time we've had available but for a 1-day course touching on both we wanted to try something a little different. For this course setup need to be easy so we didn't have to think about installing Python or git, and be able to get straight into writing Python with a bit of git thrown in as well as integrating with GitHub.
```

**Pull request merged**